test/tesla/adapter/hackney_test.exs

```diff
@@ -62,4 +62,17 @@ defmodule Tesla.Adapter.HackneyTest do
 
     assert {:error, :fake_error} = call(request)
   end
+
+  test "get with `stream_to_pid: pid` option" do
+    request = %Env{
+      method: :get,
+      url: "#{@http}/ip"
+    }
+
+    assert {:ok, %Env{} = response} = call(request, stream_to_pid: self())
```

Comment:
When passing pid explicitly, I'd expect to handle incoming message manually. 
Is there anything stopping us from using `steam_response: true` instead?

Does the implementation handle multiple concurrent requests originating from the same process?

Reply:
Hackney requires you to transfer ownership of the response to the desired PID, otherwise it will get GCed.

Workaround will be to capture `self()` during the request phase, and assume that will be the PID that will consume the steam.

Reply:
And what happens when the stream is consumed from a different process?

Reply:
From what I recall, Hackney shuts down the request after a short amount of time.

So for long running responses you get a partial reply.